libs/types/src/locations.rs

```diff
@@ -53,6 +53,14 @@ impl From<MultiLocation> for Location {
 	}
 }
 
+impl From<xcm::v2::MultiLocation> for Location {
+	fn from(ml: xcm::v2::MultiLocation) -> Self {
+		// using hash here as multilocation is significantly larger than any other enum
+		// type here -- 592 bytes, vs 40 bytes for domain address (next largest)
+		Self::XCM(BlakeTwo256::hash(&ml.encode()))
+	}
+}
```

Comment:
@wischli in `from_xcm_versioned_address_doesnt_change_if_content_stays_same`, we do:

```
let l0 = Location::from(xa.clone());
let l1 = Location::from(xb.clone());
```

where `xa` is **now** `v2::MultiLocation` and `xb` is `v3::MultiLocation`, whereas they were `v0` and `v1`, respectively. It now requires `From` impls for both `v2` and `v3`, which is what this new block of code is doing. I am not sure why we didn't need two `From` impls before (i.e, one for `v1` and one for `v2`). Please have a look 🙏

Reply:
AFAICT, we did not need to have two `From` impls as `v1::MultiLocation` and `v2::MultiLocation` seem to be exactly the same, e.g. the inner `Junction` shows no diff. That changes in v3 due to the extension of `v3::Junction` by `GlobalConsensus`.

Reply:
That makes sense! Thanks 👍